kafka-rest/src/test/java/io/confluent/kafkarest/controllers/ClusterConfigManagerImplTest.java

```diff
@@ -0,0 +1,366 @@
+
+    try {
+      clusterConfigManager.getClusterConfig(
+          CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName()).get();
+      fail();
+    } catch (ExecutionException e) {
+      assertEquals(NotFoundException.class, e.getCause().getClass());
+    }
+  }
+
+  @Test
+  public void updateClusterConfig_existingConfig_updatesConfigs() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(
+        adminClient.incrementalAlterConfigs(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                singletonList(
+                    new AlterConfigOp(
+                        new ConfigEntry(CONFIG_1.getName(), "new-value"),
+                        AlterConfigOp.OpType.SET)))))
+        .andReturn(alterConfigsResult);
+    expect(alterConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                KafkaFuture.completedFuture(null)));
+    replay(clusterManager, adminClient, alterConfigsResult);
+
+    clusterConfigManager.upsertClusterConfig(
+        CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName(), "new-value").get();
+
+    verify(adminClient);
+  }
+
+  @Test
+  public void updateClusterConfig_nonExistingCluster_throwsNotFound() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.empty()));
+    replay(clusterManager);
+
+    try {
+      clusterConfigManager.upsertClusterConfig(
+          CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName(), "new-value").get();
+      fail();
+    } catch (ExecutionException e) {
+      assertEquals(NotFoundException.class, e.getCause().getClass());
+    }
+
+    verify(clusterManager);
+  }
+
+  @Test
+  public void resetClusterConfig_existingConfig_resetsConfig() throws Exception {
+    expect(clusterManager.getCluster(CLUSTER_ID)).andReturn(completedFuture(Optional.of(CLUSTER)));
+    expect(
+        adminClient.incrementalAlterConfigs(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                singletonList(
+                    new AlterConfigOp(
+                        new ConfigEntry(CONFIG_1.getName(), /* value= */ null),
+                        AlterConfigOp.OpType.DELETE)))))
+        .andReturn(alterConfigsResult);
+    expect(alterConfigsResult.values())
+        .andReturn(
+            singletonMap(
+                new ConfigResource(ConfigResource.Type.BROKER, ""),
+                KafkaFuture.completedFuture(null)));
+    replay(clusterManager, adminClient, alterConfigsResult);
+
+    clusterConfigManager.deleteClusterConfig(
+        CLUSTER_ID, ClusterConfig.Type.BROKER, CONFIG_1.getName()).get();
+
+    verify(adminClient);
+  }
+
+  @Test
+  public void resetClusterConfig_nonExistingCluster_throwsNotFound() throws Exception {
```

Comment:
Test for resetting cluster config for a non-existing config ?

Reply:
As explained in the other comment, we can't fail if the config does not exist. It's kind of a lame behaviour which if I have time I will try to fix in AK at some point.